filesystem/filesystem.go

```diff
@@ -226,8 +252,13 @@ func (m *Mount) Setup() error {
 		return m.err(err)
 	}
 
+	dstPath, err := m.baseDirReal()
+	if err != nil {
+		return m.err(err)
+	}
+
 	// Atomically move directory into place.
-	return m.err(os.Rename(temp.BaseDir(), m.BaseDir()))
+	return m.err(os.Rename(temp.BaseDir(), dstPath))
```

Comment:
This `os.Rename` code here and below could then stay the same.

Reply:
Done